Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/xdsb/helper/XDSbAdhocQueryResponseHelperImpl.java

```diff
@@ -58,12 +58,14 @@ public String getSingleSlotValue(ResponseSlotName slotName, IdentifiableType ide
 
     @Override
     public String getSingleSlotValue(String customSlotName, IdentifiableType identifiableType) {
-        List<SlotType1> slots = identifiableType.getSlot();
-        for (SlotType1 s : slots) {
-            if (StringUtils.equalsIgnoreCase(customSlotName, s.getName()) && s.getValueList() != null) {
-                ValueListType vlt = s.getValueList();
-                for (String string : vlt.getValue()) {
-                    return string;
+        if (identifiableType != null) {
+            List<SlotType1> slots = identifiableType.getSlot();
```

Comment:
This line should be merged with the enhanced for, instead of a separate declaration.

Also, can identifiableType.getSlot() ever be null?  If so, append a check in the if statement above.